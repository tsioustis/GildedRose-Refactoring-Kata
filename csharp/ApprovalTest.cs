using System.Collections.Generic;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace csharp
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ApprovalTest
    {
        [Test]
        public void UpdateQuality_InitialState_ApprovalTest()
        {
            // Arrange
            var items = new List<Item>
            {
                new() {Name = "+5 Dexterity Vest", SellIn = 10, Quality = 20},
                new() {Name = "Aged Brie", SellIn = 2, Quality = 0},
                new() {Name = "Elixir of the Mongoose", SellIn = 5, Quality = 7},
                new() {Name = "Sulfuras, Hand of Ragnaros", SellIn = 0, Quality = 80},
                new() {Name = "Backstage passes to a TAFKAL80ETC concert", SellIn = 15, Quality = 20},
                new() {Name = "Conjured Mana Cake", SellIn = 3, Quality = 6}
            };

            var gildedRose = new GildedRose(items);

            // Act
            gildedRose.UpdateQuality();

            var updatedItemsStringBuilder = new StringBuilder();
            foreach (var item in items)
            {
                updatedItemsStringBuilder.AppendLine($"{item.Name}, {item.SellIn}, {item.Quality}");
            }

            // Assert
            Approvals.Verify(updatedItemsStringBuilder.ToString());
        }
    }
}
