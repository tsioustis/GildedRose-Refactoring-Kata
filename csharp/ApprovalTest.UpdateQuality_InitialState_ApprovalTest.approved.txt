+5 Dexterity Vest, 9, 19
Aged Brie, 1, 1
Elixir of the Mongoose, 4, 6
Sulfuras, Hand of Ragnaros, 0, 80
Backstage passes to a TAFKAL80ETC concert, 14, 21
Conjured Mana Cake, 2, 5
